src/messaging/ExchangeContext.h

```diff
@@ -199,6 +182,24 @@ class DLL_EXPORT ExchangeContext : public ReliableMessageContext,
     void Free();
     void Reset();
 
+    /**
+     *  Determine whether a response is expected for messages sent over
+     *  this exchange.
+     *
+     *  @return Returns 'true' if response expected, else 'false'.
+     */
+    bool IsResponseExpected() const;
+
+    /**
+     *  Set whether a response is expected on this exchange based on if flag kExpectResponse
+     *  is set or not in sendFlags when send a message over exchange.
```

Comment:
```suggestion
     *  Track whether we are now expecting a response to a message sent via this exchange (because that
     *  message had the kExpectResponse flag set in its sendFlags).
```